src/main/java/BlueTurtle/TSE/JSONFormatter.java

```diff
@@ -33,9 +34,22 @@
 	 */
 	public void format() throws IOException {
 		List<Warning> totalWarnings = new ArrayList<Warning>();
-		totalWarnings.addAll(parseCheckStyleXML());
-		totalWarnings.addAll(parsePMDXML());
-		totalWarnings.addAll(parseFindBugsXML());
+		
+		File file = new File(JavaController.getCheckStyleOutputFile());
+		if(file.exists() && !file.isDirectory()) { 
```

Comment:
Maybe something in the line of: 
```java

public List<Warning> parseOutPutFile(XMLParser parser, outputFile, warningList) throws IOException {
         if (checkFile(outputFile)) {
             return warningList.add(parser.parseFile(outputFile));
         }
         else {
             return warningList;
         }
}
```

Or if you want to keep it even simpler: only add if the file exist. Then you don't need the else case